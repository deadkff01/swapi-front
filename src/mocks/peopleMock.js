export const peopleMock = [
  {
    name: 'Luke Skywalker',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'http://swapi.dev/api/planets/1/',
    films: [
      'http://swapi.dev/api/films/1/',
      'http://swapi.dev/api/films/2/',
      'http://swapi.dev/api/films/3/',
      'http://swapi.dev/api/films/6/'
    ],
    species: [],
    vehicles: [
      'http://swapi.dev/api/vehicles/14/',
      'http://swapi.dev/api/vehicles/30/'
    ],
    starships: [
      'http://swapi.dev/api/starships/12/',
      'http://swapi.dev/api/starships/22/'
    ],
    created: '2014-12-09T13:50:51.644000Z',
    edited: '2014-12-20T21:17:56.891000Z',
    url: 'http://swapi.dev/api/people/1/'
  },
  {
    name: 'C-3PO',
    height: '167',
    mass: '75',
    hair_color: 'n/a',
    skin_color: 'gold',
    eye_color: 'yellow',
    birth_year: '112BBY',
    gender: 'n/a',
    homeworld: 'http://swapi.dev/api/planets/1/',
    films: [
      'http://swapi.dev/api/films/1/',
      'http://swapi.dev/api/films/2/',
      'http://swapi.dev/api/films/3/',
      'http://swapi.dev/api/films/4/',
      'http://swapi.dev/api/films/5/',
      'http://swapi.dev/api/films/6/'
    ],
    species: ['http://swapi.dev/api/species/2/'],
    vehicles: [],
    starships: [],
    created: '2014-12-10T15:10:51.357000Z',
    edited: '2014-12-20T21:17:50.309000Z',
    url: 'http://swapi.dev/api/people/2/'
  },
  {
    name: 'R2-D2',
    height: '96',
    mass: '32',
    hair_color: 'n/a',
    skin_color: 'white, blue',
    eye_color: 'red',
    birth_year: '33BBY',
    gender: 'n/a',
    homeworld: 'http://swapi.dev/api/planets/8/',
    films: [
      'http://swapi.dev/api/films/1/',
      'http://swapi.dev/api/films/2/',
      'http://swapi.dev/api/films/3/',
      'http://swapi.dev/api/films/4/',
      'http://swapi.dev/api/films/5/',
      'http://swapi.dev/api/films/6/'
    ],
    species: ['http://swapi.dev/api/species/2/'],
    vehicles: [],
    starships: [],
    created: '2014-12-10T15:11:50.376000Z',
    edited: '2014-12-20T21:17:50.311000Z',
    url: 'http://swapi.dev/api/people/3/'
  }
];

export const C3PO = {
  name: 'C-3PO',
  height: '167',
  mass: '75',
  hair_color: 'n/a',
  skin_color: 'gold',
  eye_color: 'yellow',
  birth_year: '112BBY',
  gender: 'n/a',
  homeworld: 'http://swapi.dev/api/planets/1/',
  films: [
    'http://swapi.dev/api/films/1/',
    'http://swapi.dev/api/films/2/',
    'http://swapi.dev/api/films/3/',
    'http://swapi.dev/api/films/4/',
    'http://swapi.dev/api/films/5/',
    'http://swapi.dev/api/films/6/'
  ],
  species: ['http://swapi.dev/api/species/2/'],
  vehicles: [],
  starships: [],
  created: '2014-12-10T15:10:51.357000Z',
  edited: '2014-12-20T21:17:50.309000Z',
  url: 'http://swapi.dev/api/people/2/'
};
