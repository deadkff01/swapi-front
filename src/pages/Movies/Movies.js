import { useEffect, useState, useCallback } from 'react';
import { Container } from 'components/Container';
import { MovieCard } from 'components/MovieCard';
import { useHttpInstance } from 'services/baseService';
import { Loader } from 'components/Loader';
import { Fade } from 'utils/Fade';
import { ErrorWarning } from 'components/ErrorWarning';

export default function Movies() {
  const [movies, setMovies] = useState([]);
  const { get, error, response, loading } = useHttpInstance();

  const getAllMovies = useCallback(async () => {
    const movies = await get('movies');
    if (response.ok) setMovies(movies);
  }, [get, response]);

  useEffect(() => {
    getAllMovies();
  }, [getAllMovies]);

  if (loading) {
    return <Loader />;
  }

  if (error) {
    return (
      <ErrorWarning message="Error, verify your conection or reload the page! may the force be with you!" />
    );
  }

  return (
    <Fade in={true} appear>
      <Container
        data-testid="container"
        className="flex flex-wrap justify-center mb-10"
      >
        {movies.map((m, index) => (
          <MovieCard key={index} movieInfo={m} index={index + 1} />
        ))}
      </Container>
    </Fade>
  );
}
