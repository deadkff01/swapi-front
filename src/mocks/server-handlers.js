import { rest } from 'msw';
import { mockMovies, aNewHopeMock } from './mockMovies';
import { peopleMock, C3PO } from './peopleMock';

const handlers = [
  rest.get(`${process.env.REACT_APP_ENDPOINT}/movies`, (_, res, ctx) =>
    res(ctx.json(mockMovies))
  ),
  rest.get(`${process.env.REACT_APP_ENDPOINT}/movies/1`, (_, res, ctx) =>
    res(ctx.json(aNewHopeMock))
  ),
  rest.post(`${process.env.REACT_APP_ENDPOINT}/people`, (_, res, ctx) =>
    res(ctx.json(peopleMock))
  ),
  rest.get(`${process.env.REACT_APP_ENDPOINT}/people/2`, (_, res, ctx) =>
    res(ctx.json(C3PO))
  )
];

export { handlers };
