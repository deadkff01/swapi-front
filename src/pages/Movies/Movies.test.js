import { cleanup, screen } from '@testing-library/react';
import '../../setupTests';
import { renderWithRouter } from '../../testUtils';
import Movies from './Movies';
import '@testing-library/jest-dom/extend-expect';
import { server } from 'mocks/server';

describe('Movies component', () => {
  beforeAll(() => server.listen());

  afterEach(() => server.resetHandlers());

  afterAll(() => {
    jest.useRealTimers();
    server.close();
    cleanup();
  });

  test('renders the component', async () => {
    renderWithRouter(<Movies />);
    const container = await screen.findByTestId('container');
    expect(container).toMatchSnapshot();
  });

  test('renders the loader', async () => {
    renderWithRouter(<Movies />);
    const loader = await screen.findByTestId('loader');
    expect(loader).toBeDefined();
  });

  test('movies count', async () => {
    renderWithRouter(<Movies />);
    const element = await screen.findAllByTestId('movie-card');
    expect(element.length).toBe(6);
  });
});
