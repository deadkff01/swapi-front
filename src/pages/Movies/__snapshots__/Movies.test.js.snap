// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies component renders the component 1`] = `
<div
  class="sc-bdfBwQ bqIkDZ flex flex-wrap justify-center mb-10"
  data-testid="container"
/>
`;
